src/System.IO.Ports/src/System/IO/Ports/SerialStream.cs

```diff
@@ -957,7 +954,12 @@ public unsafe override int EndRead(IAsyncResult asyncResult)
             // Free memory, GC handles.
             NativeOverlapped* overlappedPtr = afsar._overlapped;
             if (overlappedPtr != null)
-                Overlapped.Free(overlappedPtr);
+            {
+                // Legacy behavior as indicated by tests (e.g.: System.IO.Ports.Tests.SerialStream_EndRead.EndReadAfterClose)
+                // expects to be able to call EndRead after Close/Dispose - even if disposed _threadPoolBinding can free the
+                // native overlapped.
+                _threadPoolBinding.FreeNativeOverlapped(overlappedPtr);
```

Comment:
that's interesting but I don't think that means that it's defined to be safe to free it twice? if you set afsar._overlapped to null here, it might make the code more robust. In master only is fine. 
Up to you.